engine-standalone-tracing/src/types/call_tracer.rs

```diff
@@ -0,0 +1,288 @@
+                code_address,
+                transfer,
+                input,
+                target_gas,
+                is_static,
+                context,
+            } => {
+                let call_type = if is_static {
+                    CallType::StaticCall
+                } else if transfer.is_none() {
+                    CallType::DelegateCall
+                } else if code_address == context.address {
+                    CallType::Call
+                } else {
+                    CallType::CallCode
+                };
+
+                self.enter(
+                    call_type,
+                    Address::new(context.caller),
+                    Address::new(context.address),
+                    input.to_vec(),
+                    target_gas.unwrap_or_default(),
+                    context.apparent_value,
+                );
+            }
+            evm::tracing::Event::Create {
+                caller,
+                address,
+                scheme,
+                value,
+                init_code,
+                target_gas,
+            } => {
+                let call_type = match scheme {
+                    evm::CreateScheme::Legacy { .. } => CallType::Create,
+                    evm::CreateScheme::Create2 { .. } => CallType::Create2,
+                    evm::CreateScheme::Fixed(_) => CallType::Create, // is this even possible in production?
+                };
+
+                self.enter(
+                    call_type,
+                    Address::new(caller),
+                    Address::new(address),
+                    init_code.to_vec(),
+                    target_gas.unwrap_or_default(),
+                    value,
+                );
+            }
+            evm::tracing::Event::Suicide {
+                address,
+                target,
+                balance,
+            } => {
+                // TODO: gas = 0 is right?
+                self.enter(
+                    CallType::SelfDestruct,
+                    Address::new(address),
+                    Address::new(target),
+                    Vec::new(),
+                    0,
+                    balance,
+                );
+                self.exit(Vec::new(), None);
+            }
+            // Exit event always comes after RecordRefund, so we don't need to worry about gas here (it's handled in RecordRefund)
+            evm::tracing::Event::Exit {
+                reason,
+                return_value,
+            } => {
+                let error = match reason {
+                    evm::ExitReason::Succeed(_) => None,
+                    other => Some(other),
+                };
+                self.exit(return_value.to_vec(), error);
+            }
+
+            // not useful
+            evm::tracing::Event::TransactCall { .. } => (),
+            evm::tracing::Event::TransactCreate { .. } => (),
+            evm::tracing::Event::TransactCreate2 { .. } => (),
```

Comment:
Just a nit, this could be improved `_ => ()`

Reply:
I prefer explicitly listing all variants. This makes the compiler yell at you if in the future there is a new variant added that you did not consider.